great_expectations/data_context/store/checkpoint_store.py

```diff
@@ -177,22 +177,76 @@ def get_checkpoint(
         return checkpoint_config
 
     def add_checkpoint(self, checkpoint: Checkpoint) -> Checkpoint:
+        """Persist a stand-alone Checkpoint object.
+
+        Args:
+            checkpoint: The Checkpoint to persist.
+
+        Returns:
+            The persisted Checkpoint (possibly modified state based on store backend).
+
+        Raises:
+            CheckpointError: If a Checkpoint with the given name already exists.
+        """
         key = self._construct_key_from_checkpoint(checkpoint)
-        return self._add_checkpoint(key=key, checkpoint=checkpoint)
```

Comment:
Found no need to use a private here since the logic is isolated to this one func